plugins/modules/kutils.py

```diff
@@ -0,0 +1,351 @@
+  community.general.kutils:
+    password: some_password
+
+- name: Get kerberos ticket using keytab
+  community.general.kutils:
+    keytab: true
+    keytab_path: /etc/ipa/file.keytab
+
+- name: Get kerberos ticket using principal name
+  community.general.kutils:
+    password: some_password
+    principal: admin
+
+- name: Get kerberos ticket using principal with realm
+  community.general.kutils:
+    password: some_password
+    principal: admin@IPA.TEST
+
+- name: Check for existence by ticket cache
+  community.general.kutils:
+    cache_name: KEYRING:persistent:0:0
+
+- name: Make sure default ticket is destroyed
+  community.general.kutils:
+    state: absent
+
+- name: Make sure specific ticket destroyed by principal
+  community.general.kutils:
+    state: absent
+    principal: admin@IPA.TEST
+
+- name: Make sure specific ticket destroyed by cache_name
+  community.general.kutils:
+    state: absent
+    cache_name: KEYRING:persistent:0:0
+
+- name: Make sure all tickets are destroyed
+  community.general.kutils:
+    state: absent
+    kdestroy_all: true
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+class IPAKeytab(object):
+    def __init__(self, module, **kwargs):
+        self.module = module
+        self.password = kwargs['password']
+        self.principal = kwargs['principal']
+        self.state = kwargs['state']
+        self.kdestroy_all = kwargs['kdestroy_all']
+        self.cache_name = kwargs['cache_name']
+        self.start_time = kwargs['start_time']
+        self.renewable = kwargs['renewable']
+        self.forwardable = kwargs['forwardable']
+        self.proxiable = kwargs['proxiable']
+        self.address_restricted = kwargs['address_restricted']
+        self.canonicalization = kwargs['canonicalization']
+        self.enterprise = kwargs['enterprise']
+        self.renewal = kwargs['renewal']
+        self.validate = kwargs['validate']
+        self.keytab = kwargs['keytab']
+        self.keytab_path = kwargs['keytab_path']
+
+        self.kinit_executable = [module.get_bin_path('kinit', True)]
+        self.kdestroy_executable = [module.get_bin_path('kdestroy', True)]
+        self.klist_executable = [module.get_bin_path('klist', True)]
+
+        self.kinit = CmdRunner(
+            module,
+            command=self.kinit_executable,
+            arg_formats=dict(
+                lifetime=cmd_runner_fmt.as_opt_val('-l'),
+                start_time=cmd_runner_fmt.as_opt_val('-s'),
+                renewable=cmd_runner_fmt.as_opt_val('-r'),
+                forwardable=cmd_runner_fmt.as_func(self.check_for_none(self.address_restricted, '-F')),
+                proxiable=cmd_runner_fmt.as_func(self.check_for_none(self.address_restricted, '-P')),
+                address_restricted=cmd_runner_fmt.as_func(self.check_for_none(self.address_restricted, '-A')),
```

Comment:
These can be made simpler:
```suggestion
                forwardable=cmd_runner_fmt.as_bool('-f', '-F'),
                proxiable=cmd_runner_fmt.as_bool('-p', '-P'),
                address_restricted=cmd_runner_fmt.as_bool('-a', '-A'),
```
and `None` will automatically be discarded.

Reply:
Wow, thats so much better, could you please document such possibility in cmd_runner doc?

Reply:
Actually, it is documented under `cmd_runner_fmt.as_bool()` but I reckon maybe I could improve the formatting to make it clearer.

Reply:
@russoz well, it seems not to be working as expected
![image](https://github.com/user-attachments/assets/bc281ef0-96c0-4e34-b576-098b8f2c813b)

Reply:
Sorry, I barely used this situation before, I had forgotten how to handle it. What you need is:
```suggestion
                forwardable=cmd_runner_fmt.as_bool('-f', '-F', ignore_none=True),
                proxiable=cmd_runner_fmt.as_bool('-p', '-P', ignore_none=True),
                address_restricted=cmd_runner_fmt.as_bool('-a', '-A', ignore_none=True),
```
Tested locally with your branch, it works. I had to `git rebase main` to pick up the latest updates though.

Reply:
wow, that worked, thanks! But there are seems to be no mentioins of `ignore_none` in a doc https://docs.ansible.com/ansible/devel/collections/community/general/docsite/guide_cmdrunner.html 
Am I missing something?

Reply:
Nop, that one was on me. I submitted another PR, https://github.com/ansible-collections/community.general/pull/8971, to fix that. Please do review it.